tools/gdb-helpers.py

```diff
@@ -37,19 +37,30 @@
 
 
 def ats_str(addr, addr_len):
+    if addr_len == 0:
+        return ''
     #print("addr {} len {}".format(addr, addr_len))
     inferior = gdb.selected_inferior()
-    buff = inferior.read_memory(addr, addr_len)
-    if buff[0] == '\x00':
-        return 'null'
-    else:
-        return buff
+    try:
+        buff = inferior.read_memory(addr, addr_len)
+        if buff[0] == '\x00':
+            return 'null'
+        else:
+            return buff
+    except:
+        return 'unreadable({:x})'.format(int(addr))
```

Comment:
Preferable to use a format string literal:

```
        return f'unreadable({int(addr):x}'

Reply:
If you want to stick with `.format` to be compatible with more versions of Python, that's fine with me.